### ISSUE TITLE
Login should handle obscure cases

### ISSUE THREAD

**Issue by demianbrecht** [heroku/simple-salesforce#70]
https://github.com/heroku/simple-salesforce/blob/master/simple_salesforce/login.py#L128

Should this perhaps handle unexpected cases, such as unexpected server responses (invalid data)? This didn't surface in practice, I just came across it while writing some unit tests pumping in unexpected data. It's a low risk item, but just for completeness, perhaps add parsing error handling inside the above block?

**Comment by nickcatal** [comment]
Probably?

You're welcome to do a pull request and test it.

That's the part of the package I cloned from a different project. There is no RESTful way of doing username/password/secret auth with Salesforce (unless you want to create an OAuth2 app) and I built this specifically because I didn't want to learn the XML API :smile:

**Comment by demianbrecht** [comment]
Yeah, it's on the back burner for now. I'll be working on test coverage first.
